### ISSUE TITLE
Generalizability with the combine counts script 

### ISSUE THREAD

**Issue by beng1290** [whhudson/spatialTCR#4]
Hey, 

I've been working on some workflow for spatial TCR sequencing. You're paper is wonderful! Just wanted to ping some things here that I noticed about the combine.counts.py script in case anyone else is working with your code.

I didn't really know anything about fastq files when I first started working on this so maybe these things were more obvious to others. This first thing doesn't affect your workflow, but would affect anyone trying to replicate work. The second thing might cause you to lose some clone hits, but good news is that from my current testing I am still getting values in the same ballpark that you reported in your publication.

First, the code appears to be specific for your sequencer and matches read 1 to read 2 using: ```readnames.get(readname.replace("1:N:0", "2:N:0"))```. This works well for illumina sequencers where:
```
read 1: @EAS139:136:FC706VJ:2:2104:15343:197393 **1:N:18:1**
read 2: @EAS139:136:FC706VJ:2:2104:15343:197393 **2:N:18:1**
```

But when I downloaded your data from SRA the read names are replaced with SRA names:
```
read 1: @SRR001666.1
read 2: @SRR001666.2
```

Thus the script failed for me when downloading from SRA (and would fail for sequencers with other read name conventions). A better method would be to extract the read name itself and match those. 

Second, the when cycling through the clone files to capture which reads correspond to a clone; your code uses ``@`` to determine what lines correspond to read names, then capture the next line as the sequence. This leads to issues if the QC line from the fastq also starts with an ``@``, which apparently does happen. Essentially that row is also matched as a clone, and the read name for the following sequence is captured as the sequence. These "false reads" are likely dropped later when matching between read 1 and read 2 but basically you'll miss the following read. 

TLDR: I found this script that parsed fastq files very quickly, essentially as sets of 4 and returned the read name, sequence, and QC line as tuples. Using that instead solved the issues and was faster than the code you had when benchmarking. Would highly recommend. Here's a link: https://github.com/lh3/readfq. I'm working on some code to formalize this step but don't have much to share at this point.
